tachyon/zk/plonk/lookup/lookup_argument_runner_impl.h

```diff
@@ -56,6 +57,54 @@ LookupPermuted<Poly, Evals> LookupArgumentRunner<Poly, Evals>::PermuteArgument(
           std::move(permuted_input_poly), std::move(permuted_table_poly)};
 }
 
+template <typename Poly, typename Evals>
+template <typename PCSTy, typename F>
+LookupCommitted<Poly> LookupArgumentRunner<Poly, Evals>::CommitPermuted(
+    Prover<PCSTy>* prover, const LookupPermuted<Poly, Evals>& permuted,
+    const F& beta, const F& gamma) {
+  BlindedPolynomial<Poly> grand_product_poly = GrandProductArgument::Commit(
+      prover, CreateNumeratorCallback<F>(permuted, beta, gamma),
+      CreateDenominatorCallback<F>(permuted, beta, gamma));
+
+  return LookupCommitted<Poly>(std::move(permuted.permuted_input_poly()),
+                               std::move(permuted.permuted_table_poly()),
```

Comment:
```suggestion
  return LookupCommitted<Poly>(std::move(permuted).permuted_input_poly()),
                               std::move(permuted).permuted_table_poly()),
```

With this modificaiton!

```c++
  BlindedPolynomial<Poly>&& permuted_input_poly() && {
    return std::move(permuted_input_poly_);
  }
```